codedeploy_role/main.tf

```diff
@@ -0,0 +1,47 @@
+
+resource "aws_iam_role_policy" "codedeploy_policy" {
+    name = "codedeploy"
+    role = "${aws_iam_role.codedeploy_role.id}"
+    policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "autoscaling:CompleteLifecycleAction",
+                "autoscaling:DeleteLifecycleHook",
+                "autoscaling:DescribeAutoScalingGroups",
+                "autoscaling:DescribeLifecycleHooks",
+                "autoscaling:PutLifecycleHook",
+                "autoscaling:RecordLifecycleActionHeartbeat",
+                "ec2:DescribeInstances",
+                "ec2:DescribeInstanceStatus",
+                "tag:GetTags",
+            "tag:GetResources"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+EOF
+}
+
+resource "aws_iam_role" "codedeploy_role" {
+    name = "codedeploy-role"
+    assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "codedeploy.eu-west-1.amazonaws.com"
```

Comment:
or a data source? I think there was something added to get the current region: https://www.terraform.io/docs/providers/aws/d/region.html